backend/src/test/java/com/github/devbk84/backend/controller/ShopControllerTest.java

```diff
@@ -0,0 +1,42 @@
+package com.github.devbk84.backend.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.devbk84.backend.repo.ShopRepo;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.web.context.WebApplicationContext;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+@SpringBootTest
+@AutoConfigureMockMvc
+class ShopControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private WebApplicationContext context;
+
+    @Autowired
+    private ShopRepo shopRepo;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+
+
+    @Test
+    @DirtiesContext
+    void getAllProducts() throws Exception {
+        mockMvc.perform(get("/api/products"))
+                .andExpect(status().isOk())
+                .andExpect(content().json("[]"));
+    }
```

Comment:
You might want to describe the expected scenario in the test's name.
Good use of new lines though :+1: Gives a good structure